python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -669,13 +763,94 @@ def stream(
                 def my_dbt_assets(context, dbt: DbtCliResource):
                     yield from dbt.cli(["run"], context=context).stream()
         """
-        for event in self.stream_raw_events():
-            yield from event.to_default_asset_events(
-                manifest=self.manifest,
-                dagster_dbt_translator=self.dagster_dbt_translator,
-                context=self.context,
-                target_path=self.target_path,
-            )
+        # We keep a list of emitted Dagster events and pending futures which augment
+        # emitted events with additional metadata. This ensures we can yield events in the order
+        # they are emitted by dbt.
+        output_events_and_futures: List[Union[Future, DbtDagsterEventType]] = []
+
+        with pushd(str(self.project_dir)):
+            with ThreadPoolExecutor(max_workers=self.stream_events_threadpool_size) as executor:
```

Comment:
nit: you can combine contextmanagers in a single with block so we don't have excessive nesting.